### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by JakubJT** [qwtel/sqlite-viewer-vscode#13] [Q&A] [answered]
Under what license is the sqlite-viewer-vscode released?

**Comment by qwtel** [comment, marked as answer]
Use of the extension is guided by the vscode marketplace terms of use.

The code remains closed and I reserve all rights. If I ever get around to tidying it up might publish it under an open source license.

Parts of the source code are licensed separately, which can be found here https://github.com/qwtel/sqlite-viewer-vscode/blob/master/LICENSE.md

**Reply by zypA13510** [reply]
> ### Microsoft Visual Studio Marketplace Terms of Use 
> Last Updated June 2021
> 
> #### 1. Use Rights
> a. **Publisher Terms of Use**. Your right to use any Offering is governed by separate terms of use 
> provided by the Publisher of the Offering (**“Publisher Terms of Use”**). Except for Offerings that we 
> publish, we are not a party to and are not bound by any Publisher Terms of Use. You are solely 
> responsible for your dealings with the Publisher.

IANAL. But what I see from the text above is that the Visual Studio Marketplace ToU does not grant any usage rights of the extension to the user. So for users to be able to use your extension legally, you would need to explicitly grant us the right to use it through an appropriate license.

Btw, publishing it under an open-source license does not prevent you from tidying it up, you can always publish an unstable version like 0.0.1 and improve it in later versions - unless you have proprietary code in this repo (but then it would just confirm that we cannot use it until it is cleared of the proprietary code).
